misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt

```diff
@@ -46,6 +46,10 @@ internal class SessionFactoryService(
 ) : AbstractIdleService(), Provider<SessionFactory>, TransacterService {
   private var sessionFactory: SessionFactory? = null
 
+  val threadInTransaction = object : ThreadLocal<Boolean>() {
+    override fun initialValue() = false
+  }
```

Comment:
Much of the other changes in this PR are so we can put the in-transaction ThreadLocal here on the SessionFactoryService. It’s 1:1 with the database!